### PR TITLE
[SYCL] Enable level-zero-link-flags and fix kernel_bundle_api

### PR DESCRIPTION
**Pull request by HabKaffee**
This patch enabling KernelAndProgram/level-zero-link-flags.cpp due to https://github.com/intel/llvm/pull/5476

### CODE REVIEW

File: SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp [hunk 1]
```diff
@@ -1,8 +1,7 @@
-// Disable fallback assert here until online-support is fixed.
 // Use of per-kernel device code split and linking the bundle with all images
 // involved leads to multiple definition of AssertHappened structure due each
 // device image is statically linked against fallback libdevice.
-// RUN: %clangxx -DSYCL_DISABLE_FALLBACK_ASSERT=1 -fsycl -fsycl-device-code-split=per_kernel -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %clangxx -fsycl -fsycl-device-code-split=per_kernel -fsycl-targets=%sycl_triple %s -o %t.out
```
Comment (gmlueck) [inline SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp]: I'm curious, how is this related to the bug that was fixed in intel/llvm#5476?

Comment (HabKaffee) [reply SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp]: This test started failing after I started check https://github.com/intel/llvm/pull/5476 . Moreover, there was a comment line ` //Disable fallback assert here until online-support is fixed.` I can guess, that it is about fixed bug.

Comment (HabKaffee) [reply SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp]: I resolved the problem. `Basic/kernel_bundle/kernel_bundle_api`  was affected indirectly. Before https://github.com/intel/llvm/pull/5476 it received a nullptr as a link option, but now here is passed an empty string.

File: SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp [hunk 2]
```diff
@@ -1,4 +1,3 @@
-// Disable fallback assert here until online-support is fixed.
```
Comment (gmlueck) [inline SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp:1]: It might be worth asking @s-kanaev if this comment should be removed.  It looks to me like the test still has fallback assert disabled, so maybe the comment should remain until it is enabled.

Comment (HabKaffee) [reply SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp:1]: I'll ask about your concern.

**Comment by HabKaffee** [comment]
https://github.com/intel/llvm/pull/5476

**Comment by HabKaffee** [comment]
https://github.com/intel/llvm/pull/5476

**Comment by HabKaffee** [comment]
https://github.com/intel/llvm/pull/5476

File: SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp (hunk 2 above)
Comment (s-kanaev) [reply SYCL/Basic/kernel_bundle/kernel_bundle_api.cpp:1]: Fallback assert is now disabled by default. See https://github.com/intel/llvm/pull/4694.
Hence, this comment might be removed along with `-DSYCL_DISABLE_FALLBACK_ASSERT=1` in build line.
It'll be responsibility of the one who enables the fallback assert to verify it won't impact other tests.

P.S.: Sorry for late response.

**Comment by HabKaffee** [comment]
@s-kanaev @alexbatashev Could you, please, take a look on this PR?

**Review by s-kanaev — APPROVED** [review]
The change seems legit.

**Review by vladimirlaz — APPROVED** [review]
